README.md

```diff
@@ -1,3 +1,22 @@
 # CA$H€R
 
-![](http://4.bp.blogspot.com/_XdP6Lp2ceqY/TEJuww2sk2I/AAAAAAAAWpY/EmWweRXahGM/s1600/tumblr_l55qcmER041qznd83o1_500.jpg)
+Casher is a Ruby script used by [Travis Build](https://github.com/travis-ci/travis-build) to fetch, create, and update caches. Casher fetches a cache if one exists and creates one if it doesn't. It uses md5deep to detect changes in an existing cache and updates the cache if a difference in the MD5 checksums are found. Casher's source is hosted and used directly from Github.
+
+![Comic about cash](http://4.bp.blogspot.com/_XdP6Lp2ceqY/TEJuww2sk2I/AAAAAAAAWpY/EmWweRXahGM/s1600/tumblr_l55qcmER041qznd83o1_500.jpg)
+
+## Where it fits in
+
+[Travis Build](https://github.com/travis-ci/travis-build) generates a signature and URL for a cache and sends it to Casher. Casher checks the URL to see if there is an existing cache. If one is not found, it creates one. If one is found, it pulls the cache from S3 (Docker builds) or GCS (GCE and MacStadium builds) and expands it from the root of the OS. 
```

Comment:
This feels a bit insufficient.

`travis-build` typically generates a few URLs (which include AWS-compatible signatures) and pass them to `casher`. `casher` then goes through this list in order of preference (which is exactly the order `travis-build` gives them in). It will stop when one is found.

As I said elsewhere, it is not important to note where these URLs point to. This is a `travis-build` concern; `casher` simply accepts these URLs and try.